plugins/omar-wms-plugin/grails-app/services/omar/wms/WebMappingService.groovy

```diff
@@ -275,32 +276,35 @@ class WebMappingService implements InitializingBean
     }
   }
 
-  def getMap(GetMapRequest wmsParams)
+  def getMap(GetMapRequest wmsParams, Boolean getPsm = false)
   {
     def requestType = "GET"
-    def requestMethod = "GetMap"
-    def OPTIMIZED_FORMAT = "image/vnd.jpeg-png"
+    def requestMethod = getPsm ? "GetPsm" : "GetMap"
+
     Date startTime = new Date()
     def responseTime
-    def requestInfoLog
     def httpStatus
     def filename
     def bboxMidpoint
     def result
-    Boolean addLocation = true
 
     Map<String, Object> omsParams = parseLayers( wmsParams )
 
     if ( omsParams )
     {
+      if (getPsm && ! omsParams.hasProperty('images[1].file')) {
+          throw new Exception('The filter given to the getPsm call did not contain at least two images')
```

Comment:
how does this exception get handled?

Reply:
I suspect Grails catches this and throws back an error in the response?